packages/connect/src/SmoldotProvider/SmoldotProvider.test.ts

```diff
@@ -131,6 +131,20 @@ test('emits connect and never emits disconnect for development chain', async ()
   });
 }, 10000);
 
+test('emits events when Grandpa finishes sync and then connects', async () => {
```

Comment:
@raoulmillais This test checks (based on health responses) that Grandpa initially hasn't finished sync while on 2nd health response syncing is finished (isSyncing: false).

Even though test seems to pass:
```
  ✓ emits events when Grandpa finishes sync and then connects (2 ms)
```
I receive the following log in the console:
```
/.../repos/parity/substrate-connect/packages/connect/node_modules/@substrate/smoldot-test-utils/dist/index.js:86
            throw new Error(NO_MORE_RESPONSES);
            ^

Error: json_rpc_callback was called but there are no mock responses left to respond with
    at /.../repos/parity/substrate-connect/packages/connect/node_modules/@substrate/smoldot-test-utils/src/index.ts:123:13
    at /.../repos/parity/substrate-connect/packages/connect/node_modules/@substrate/smoldot-test-utils/src/index.ts:171:35
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
    at runNextTicks (internal/process/task_queues.js:62:3)
    at listOnTimeout (internal/timers.js:523:9)
    at processTimers (internal/timers.js:497:7)
```

I am sure I am missing some point here on why test pass but this appears. Can you please support?

Reply:
This means that the fake smoldot was asked to respond to more health requests than there were mock responses.  I.E. it was asked at least 3 times - you supplied 2 mock responses.  

This `provider.healthPingerInterval = 1;` makes the `setInterval` timeout 1ms for the healthpinger effectively making it send a health request for every turn of the event loop.  It's probable that there are more turns of the event loop than you were expecting when the `SmoldotProvider` internally calls `start` on the fake smoldot and each time `sendJsonRpc` is called (each request is [processed on a new turn of the event loop](https://github.com/paritytech/substrate-connect/blob/7ccf3ba525dd3b91c7decc4fd7a4a73cdaf2dcff/packages/smoldot-test-utils/src/index.ts#L168) to avoid missing asynchrony bugs)

Reply:
Understood. I'm not sure though why the tests pass while there is an log that denotes that nothing was responded. I tried to add extra responses but test still pass and error still appear.